kore/src/SMT/SimpleSMT.hs

```diff
@@ -332,22 +335,29 @@ newSolver ::
     Logger ->
     IO SolverHandle
 newSolver exe opts logger = do
-    (hIn, hOut, hErr, hProc) <- runInteractiveProcess exe opts Nothing Nothing
+    (Just hIn, Just hOut, Just hErr, hProc) <- createProcess solverProcess
     let solverHandle = SolverHandle{hIn, hOut, hErr, hProc, queryCounter = 0}
         solver = Solver{solverHandle, logger}
 
-    _ <- forkIO $ do
-        let handler X.SomeException{} = return ()
-        X.handle handler $
-            Monad.forever $ do
-                errs <- Text.hGetLine hErr
-                debug (Solver solverHandle logger) errs
+    -- redirect error pipe to debug logger
+    Text.hGetLine hErr >>= debug (Solver solverHandle logger)
+        & Monad.forever
+        & X.handle (\X.SomeException{} -> return ())
+        & forkIO
+        & void
```

Comment:
It took me a while to realize that this just refactors the old code. I think the old code is clearer, I'd probably just factor out the `Monad.forever $ do ...` part and name it `redirectErrorsToLogger`, and also rename the `handler` to `ignoreExceptions` or something of the sort.

The main part of the code will look like:
```suggestion
  void $ forkIO (X.handle ignoreExceptions redirectErrorsToLogger)
```

This way the code is self documenting, removing any need of additional comments.

Reply:
Ok, updated in https://github.com/kframework/kore/pull/2900/commits/595b7c2522e525ce9b87870a37c47130c3b299a3. Let me know what you think.